### ISSUE TITLE
Build Fails w/ LIBPLACEBO v306 - OK w/ v302

### ISSUE THREAD

**Issue by frkli1965** [mpv-player/mpv#12104]
- mpv version: v0.36.0-59-gf6de44dd6a
- Linux Distribution and Version: Ubuntu 22.04 Kernel v6.5.0rc5
- Window Manager and version: Hyprland - Wayland git/master
- GPU driver and version: Kisak (latest)

Meson Build Options:

-Dbuildtype=release -Dcdda=enabled -Ddvdnav=enabled -Dopenal=enabled -Dwayland=enabled -Dlibmpv=true

Trace (with symbolic link libplacebo.so targeting  libplacebo.so.306 git/master) is as follows:-

/usr/local/bin/ld: libmpv.so.2.1.0.p/video_out_placebo_utils.c.o: in function `mppl_log_create':
utils.c:(.text+0x95): undefined reference to `pl_log_create_302'
collect2: error: ld returned 1 exit status
[46/46] Linking target libmpv.so.2.1.0
ninja: build stopped: subcommand failed.

With symbolic link libplacebo.so targeting libplacebo.so.302 all is well...

Suspect this has something to do with commit #f6de44d.

I appreciate that I'm a bit ahead of the curve here but thought I'd give you the heads-up nevertheless as It's bound to cause problems in due course...

Kind regards,

**Comment by llyyr** [comment]
This is unrelated to that commit.

try `meson setup build --wipe` then try rebuilding

**Comment by frkli1965** [comment]
A word to the wise...

After further investigation I've finally worked out what the problem was so here's a cautionary tale for the unwary.

If FFMPEG has libplacebo enabled in its build options and is compiled with a different version of libplacebo to that of MPV then MPV will either fail to build as in my case or fail to run should the version of libplacebo that FFMPEG was built against is missing.

Interestingly enough, should both MPv's version of libplacebo and FFMPEG's version of libplacebo be present then MPV will build and operate but who knows what strange goings-on may result.

**Comment by haasn** [comment]
^ this works fine on my system though:

```
λ ldd `which mpv` | grep placebo
	libplacebo.so.308 => /home/nand/dev/prefix//lib64/libplacebo.so.308 (0x00007f453f3f0000)
	libplacebo.so.264 => /lib64/libplacebo.so.264 (0x00007f453bcc2000)
```

mpv is compiled against libplacebo 308, ffmpeg against 264.

**Comment by frkli1965** [comment]
> ^ this works fine on my system though:
> 
> ```
> λ ldd `which mpv` | grep placebo
> 	libplacebo.so.308 => /home/nand/dev/prefix//lib64/libplacebo.so.308 (0x00007f453f3f0000)
> 	libplacebo.so.264 => /lib64/libplacebo.so.264 (0x00007f453bcc2000)
> ```
> 
> mpv is compiled against libplacebo 308, ffmpeg against 264.

Yes, it does work which was what I was trying to say, perhaps not very well.

The problem I had was that I compiled FFMPEG with v302 previously and then updated libplacebo afterwards to v306. Then without thinking, I uninstalled v302 believing that it was no longer required having forgotten entirely about FFMPEG's dependencies.

Then I proceeded to compile MPV which although picking up that v306 was installed, complained that it couldn't find v302 which was perfectly understandable with hindsight and there lies the problem.

Having totally forgotten about FFMPEG's requirements, I was at a complete loss as to why MPV was still dependent on v302 when v306 was the only libplacebo installed. What was even more peculiar to me at the time, I then went on to v308 and then MPV complained that it couldn't find v306. LOL !

It seemed that it was always dependent on the previous version, then I finally remembered that I had enabled libplacebo on the FFMPEG build and that MPV was really complaining on behalf of FFMPEG as I had yet again removed the previous version of libplacebo.

Anyway, what I was really querying with my previous message is what (if any) are the consequences of having MPV and FFMPEG reliant on varying versions of libplacebo?

Wouldn't such a scenario give rise to unpredictable behaviour potentially?

Just curious that's all, as I am no way knowledgeable enough to deduce that for myself. Logic would suggest to me that it's not an  ideal situation although I am certain that you're far more qualified than me as to whether that's indeed the case or not.

All the best... :-)
